src/main/java/org/prebid/server/analytics/AnalyticsReporterDelegator.java

```diff
@@ -168,4 +181,43 @@ private static ObjectNode prepareAnalytics(ObjectNode analytics, String adapterN
 
         return !analyticsNodeCopy.isEmpty() ? analyticsNodeCopy : null;
     }
+
+    private <T> void processEventWithResult(T event, AnalyticsReporter analyticsReporter) {
+        final String analyticsCode = analyticsReporter.name();
+        analyticsReporter.processEvent(event)
+                .compose(s -> processSuccess(s, analyticsCode))
+                .recover(s -> processFail(s, event, analyticsCode));
```

Comment:
1. You may use `otherwise` instead of `recover` since you don't have to return any result.
2. Rename `s` to `exception`.